roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,36 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set gateway address for multi subnet 
+  set_fact: dfl_nw_gw="{{ item.gateway }}"
+  when: (update_routes|default('false')|bool) and ((ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') ))) 
+  with_items: "{{ undercloud_cidr }}"
```

Comment:
Just to clarify,
If undercloud_cidr:
         - cidr: 172.20.10.0/24

and the ursula was run with -e update_routes=yes, the task "add static route for multi subnet" will fail trying to replace the route on the OS since gateway is not specified.

If we do not fail at this point, the env will deployed but the subsequent nova tasks would fail because the routes are not set up between the 2 subnets.